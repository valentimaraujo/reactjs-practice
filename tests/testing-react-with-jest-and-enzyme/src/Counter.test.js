import React from 'react';
import Counter from './Counter';

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('Counter Testing', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('start with a count of 0', () => {
    const wrapper = shallow(<Counter />);
    const text = wrapper.find('#title').text();
    expect(text).toEqual('Current count: 0');
  });

  it('can increment the count when the button is clicked', () => {
    const wrapper = shallow(<Counter />);
    const incrementBtn = wrapper.find('button.increment');
    incrementBtn.simulate('click');
    const text = wrapper.find('#title').text();
    expect(text).toEqual('Current count: 1');
  });

  it('can decrement the count when the button is clicked', () => {
    const wrapper = shallow(<Counter />);
    const decrementtBtn = wrapper.find('button.decrement');
    decrementtBtn.simulate('click');
    const text = wrapper.find('#title').text();
    expect(text).toEqual('Current count: -1');
  });
});
