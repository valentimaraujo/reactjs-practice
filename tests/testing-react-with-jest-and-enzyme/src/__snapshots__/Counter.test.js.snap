// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Testing matches the snapshot 1`] = `
<div
  className="App"
>
  <p
    id="title"
  >
    Current count: 
    0
  </p>
  <button
    className="increment"
    onClick={[Function]}
  >
    INCREMENT COUNT
  </button>
  <button
    className="decrement"
    onClick={[Function]}
  >
    DECREMENT COUNT
  </button>
</div>
`;
